swift.go

```diff
@@ -70,48 +79,74 @@ func (s *SwiftContainer) Has(k ds.Key) (bool, error) {
 		return false, err
 	}
 }
+
+func (s *SwiftContainer) GetSize(k ds.Key) (int, error) {
+	info, _, err := s.conn.Object(s.Container, keyToName(k))
+
+	if err != nil {
+		switch err {
+		case swift.ObjectNotFound:
+			return 0, ds.ErrNotFound
+		default:
+			return 0, err
+		}
+	}
+
+	maxInt := int64((^uint(0)) >> 1)
+	if info.Bytes > maxInt {
+		return 0, fmt.Errorf("integer overflow")
+	}
+	return int(info.Bytes), nil
+}
+
 func (s *SwiftContainer) Query(q dsq.Query) (dsq.Results, error) {
 	opts := swift.ObjectsOpts{
-		Prefix: q.Prefix,
-		Limit: q.Limit + q.Offset,
+		Prefix: strings.TrimPrefix(q.Prefix, "/"),
 	}
 
-	objs, err := s.conn.Objects(s.Container, &opts)
-	if err != nil {
-		return nil, err
+	if q.Limit != 0 {
+		opts.Limit = q.Limit + q.Offset
 	}
 
 	if q.Orders != nil || q.Filters != nil {
 		return nil, fmt.Errorf("swiftds doesnt support filters or orders")
 	}
 
-	res := make([]dsq.Entry, len(objs[q.Offset:]))
-	for i, obj := range objs[q.Offset:] {
-		res[i] = dsq.Entry{Key: obj.Name}
+	names, err := s.conn.ObjectNamesAll(s.Container, &opts)
```

Comment:
Is it possible to make use of `Marker` / `KeepMarker` to avoid having the query request all items from the first - Offset + Limit? Even if I have a relatively small Limit, If my query pages through all results, the final request would itself re-fetch the entire list of items, which seems pretty inefficient.